docusaurus/docs/iOS/components/message-list.md

```diff
@@ -1,3 +1,266 @@
+            messageComposerVC.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            messageComposerVC.view.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            messageComposerVC.view.topAnchor.constraint(equalTo: messageListVC.view.bottomAnchor),
+            messageComposerVC.view.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+
+        // Setup
+        messageListVC.delegate = self
+        messageListVC.dataSource = self
+        messageListVC.client = channelController.client
+
+        messageComposerVC.channelController = channelController
+        messageComposerVC.userSearchController = userSuggestionSearchController
+
+        channelController.setDelegate(self)
+        channelController.synchronize( { [weak self] _ in
+            self?.messageComposerVC.updateContent()
+        })
+    }
+}
+```
+
+:::tip 
+In order to be easier to setup child view controllers you can add this extension to your application:
+
+```swift
+extension UIViewController {
+    func addChildViewController(_ child: UIViewController, targetView superview: UIView) {
+        child.willMove(toParent: self)
+        addChild(child)
+        superview.addSubview(child.view)
+        child.didMove(toParent: self)
+    }
+}
+```
+:::
+
+Next, we need to implement the `ChatMessageListVCDataSource` to connect the messages from the `ChannelController` to the `ChatMessageListVC`.
+```swift
+extension ChannelViewController: ChatMessageListVCDataSource {
+    open func channel(for vc: ChatMessageListVC) -> ChatChannel? {
+        channelController.channel
+    }
+
+    open func numberOfMessages(in vc: ChatMessageListVC) -> Int {
+        channelController.messages.count
+    }
+
+    open func chatMessageListVC(_ vc: ChatMessageListVC, messageAt indexPath: IndexPath) -> ChatMessage? {
+        return channelController.messages[indexPath.item]
+    }
+
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
+        messageLayoutOptionsAt indexPath: IndexPath
+    ) -> ChatMessageLayoutOptions {
+        guard let channel = channelController.channel else { return [] }
+        return components.messageLayoutOptionsResolver.optionsForMessage(
+            at: indexPath,
+            in: channel,
+            with: AnyRandomAccessCollection(channelController.messages),
+            appearance: appearance
+        )
+    }
+}
+```
+As you can see, we implement the data source to feed the message list with the messages from the `ChannelController`. The most complex part is to calculate the layout options for each message, but since our custom `ChannelViewController` implements the `ThemeProvider` protocol, we have access to the `components` and `appearance` config, and with these we can use the `MessageLayoutOptionsResolver` to calculate the layout options for us.
+
+The `ChatMessageListVC` provides events throught the `ChatMessageListVCDelegate` so that we can react to actions and important lifecycle events.
```

Comment:
```suggestion
The `ChatMessageListVC` provides events through the `ChatMessageListVCDelegate` so that we can react to actions and important lifecycle events.
```